main/src/cgeo/geocaching/location/DistanceParser.java

```diff
@@ -58,15 +58,28 @@ public static float parseDistance(final String distanceText, final boolean metri
         return convertDistance(value, unit);
     }
 
+
+    /**
+     * Parses a distance string {@code distanceText} representing distance in units {@code unit}
+     *
+     * @param distanceText distance string
+     * @param unit         unit to convert from
+     * @return the distance in kilometers
+     * @throws NumberFormatException if the given string is not numeric
+     */
+    public static float parseDistance(final String distanceText, final DistanceUnit unit)
```

Comment:
So far so good.

Can you replace the calls to the other `parseDistance` method by calls to this method?
Afterwards, you can inline the `convertDistance` method.

Reply:
I replaced both occurrences connected with activity with unit selector. The other two occurrences are in GCParser and there this method is being used for parsing strings from HTML. 
When I agree that it can make sense to replace it, however it will require something like "parseUnit" method, since there is no "selector" in bare string and I think it goes beyond initial #6376 issue.

Reply:
done, however I still think that it doesn't fit in this PR fixing #6376